### ISSUE TITLE
Bug Report: After setting a topic of a message broker within a service to receive or send you cant set other topics that belong to the message broker

### ISSUE THREAD

**Issue by GreenMachine01** [amplication/amplication#6771]

**Comment by Sheersendu** [comment]
@GreenMachine01 I would like to work on this issue can it be kindly assigned to me please?

**Comment by GreenMachine01** [comment]
@Sheersendu assigned to you. We are interested in releasing the fix for this one as soon as possible. If you are unable to solve it soon please let me know.

**Comment by Sheersendu** [comment]
HI @GreenMachine01 , I followed your issue replication steps but I didn't face the bug : I'm able to set the second topic just after setting the first topic (only once). Is there any other way to replicate this issue?

**Comment by GreenMachine01** [comment]
@Sheersendu Ill try to recreate it on my side and update you.

**Comment by Sheersendu** [comment]
@GreenMachine01  yup i tried once again I got the issue, checking why the reason why.

**Comment by GreenMachine01** [comment]
@Sheersendu It just happened to me on a new service.
select the first topic to be 'send'

**Comment by Sheersendu** [comment]
Even sometimes the first topic is even not getting set I see, looking into it

> @Sheersendu It just happened to me on a new service. select the first topic to be 'send'

**Comment by Sheersendu** [comment]
Hey @GreenMachine01 , I have found a working solution but I'm stuck in one place where we need to update the memo used in TopicsList.tsx as the pattern value is changing. But its not possible to update memo, tried to use  a combination of useState along with useMemo still it didn't work, mysteriously some keys stored in memo is getting deleted. Open for further discussion or any possible solution if known. I've have spend 5-6 hours on this and really want to solve this.

**Comment by GreenMachine01** [comment]
@overbit or @abrl91 can you assist?

**Comment by Sheersendu** [comment]
@GreenMachine01 I've created PR, could you please review it?

**Comment by GreenMachine01** [comment]
@Sheersendu thank you! the team will check it out.

**Comment by Sheersendu** [comment]
Hey @GreenMachine01, can you please let me know the status of this issue. Its under review I can see, but it also says changes requested, kindly let me know what should be my follow up steps. Thanks 😄

**Comment by GreenMachine01** [comment]
@Sheersendu thanks for checking in. I will ping the team to see where it is standing.

**Comment by overbit** [comment]
Issue move to ready for QA

**Comment by morhag90** [comment]
Doesn't pass QA.
Now have an issue with the first topic. 
The screen recording is attached. 

https://github.com/amplication/amplication/assets/97830649/1c2ce931-7f07-4a1e-8a26-7380e65a0792

**Comment by Sheersendu** [comment]
> Doesn't pass QA. Now have an issue with the first topic. The screen recording is attached.
> Screen.Recording.2023-10-03.at.12.33.48.mov

Issue noted. Works on my machine, trying to repro the bug. Will keep posted!

**Comment by overbit** [comment]
Possibly related to #7324
